backend/routes/auth.py

```
@@ -0,0 +1,276 @@
+
+        return {
+            "success": True,
+            "session_id": auth_result["session_id"],
+            "user": {"username": user_data["username"], "authenticated": True},
+            "message": "Login successful",
+        }
+
+    except HTTPException:
+        raise
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Login failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)